packages/react-component-library/src/components/TextInput/partials/StyledAdornment.tsx

```diff
@@ -0,0 +1,37 @@
+import styled, { css, FlattenSimpleInterpolation } from 'styled-components'
+import { selectors } from '@royalnavy/design-tokens'
+
+export type AdornmentPositionType = 'start' | 'end'
+
+interface StyledAdornmentProps {
+  $position: AdornmentPositionType
+}
+
+const { spacing } = selectors
+
+function getPositionStyles(
```

Comment:
Another option is that you just have the `styles` `object` and then call that from `StyledAdornment`.